plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -240,6 +245,36 @@ func compareTopics(expected []string, incoming []string) bool {
 }
 
 func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Message) error {
+	var remainingLength int
+	var qosFlagsSize int
+	var qOsSize int
+	topicSize := len(msg.Topic()) + 2
+	if int(msg.Qos()) == 1 || int(msg.Qos()) == 2 {
+		qOsSize = 2
+	} else {
+		qOsSize = 0
+	}
+	payloadSize := len(msg.Payload())
+	remainingContent := topicSize + qOsSize + payloadSize
+	if remainingContent <= 127 {
+		remainingLength = 1
+	} else if remainingContent <= 16383 {
+		remainingLength = 2
+	} else if remainingContent <= 2097151 {
+		remainingLength = 3
+	} else {
+		remainingLength = 4
+	}
+	publishMessageSize := remainingLength + remainingContent + 1
+	if int(msg.Qos()) == 1 {
+		qosFlagsSize = 4
+	} else if int(msg.Qos()) == 2 {
+		qosFlagsSize = 12
+	}
```

Comment:
Can we maybe unify the two `Qos` if statements to
```suggestion
	switch int(msg.Qos()) {
	case 1: 
		qOsSize = 2
		qosFlagsSize = 4
	case 2:
		qOsSize = 2
		qosFlagsSize = 12
	default
		qOsSize = 0
	}
	payloadSize := len(msg.Payload())
	remainingContent := topicSize + qOsSize + payloadSize
	if remainingContent <= 127 {
		remainingLength = 1
	} else if remainingContent <= 16383 {
		remainingLength = 2
	} else if remainingContent <= 2097151 {
		remainingLength = 3
	} else {
		remainingLength = 4
	}
	publishMessageSize := remainingLength + remainingContent + 1
```